test/e2e/vmagent_test.go

```diff
@@ -71,9 +73,134 @@ func TestVMAgent(t *testing.T) {
+}
+
+func testExternalNodeSupportBundleCollection(t *testing.T, data *TestData, vmList []vmInfo) {
+	sftpServiceYAML := "sftp-deployment.yml"
+	secretUserName := "foo"
+	secretPassword := "pass"
+	uploadFolder := "upload"
+	uploadPath := path.Join("/home", secretUserName, uploadFolder)
+	secretName := "support-bundle-secret"
+	vmNames := make([]string, 0, len(vmList))
+	for _, vm := range vmList {
+		vmNames = append(vmNames, vm.nodeName)
+	}
+	code, _, stderr, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s -n %s", sftpServiceYAML, data.testNamespace))
+	if code != 0 {
+		t.Errorf("Error When applying %s: %v", sftpServiceYAML, stderr)
+	}
+	if err != nil {
+		t.Errorf("Error When applying %s: %v", sftpServiceYAML, err)
+	}
+	failOnError(data.waitForDeploymentRealized(t, data.testNamespace, "sftp", 30*time.Second), t)
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+		},
+		Data: map[string][]byte{
+			"username": []byte(secretUserName),
+			"password": []byte(secretPassword),
+		},
+	}
+	_, err = data.clientset.CoreV1().Secrets(namespace).Create(context.TODO(), sec, metav1.CreateOptions{})
+	require.NoError(t, err)
+	bundleName := "support-bundle-collection-external-node"
+	sc := &crdv1alpha1.SupportBundleCollection{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "support-bundle-collection-external-node",
+		},
+		Spec: crdv1alpha1.SupportBundleCollectionSpec{
+			ExternalNodes: &crdv1alpha1.BundleExternalNodes{
+				NodeNames:    vmNames,
+				NodeSelector: &metav1.LabelSelector{},
+				Namespace:    namespace,
+			},
+			ExpirationMinutes: 300,
+			FileServer: crdv1alpha1.BundleFileServer{
+				URL: fmt.Sprintf("%s:30010/upload", controlPlaneNodeIPv4()),
+			},
+			Authentication: crdv1alpha1.BundleServerAuthConfiguration{
+				AuthType: "BasicAuthentication",
+				AuthSecret: &v1.SecretReference{
+					Name:      secretName,
+					Namespace: namespace,
+				},
+			},
+		},
+	}
+	_, err = data.crdClient.CrdV1alpha1().SupportBundleCollections().Create(context.TODO(), sc, metav1.CreateOptions{})
+	require.NoError(t, err)
+	failOnError(data.waitForSupportBundleCollectionRealized(t, bundleName, 30*time.Second), t)
+	pods, err := data.clientset.CoreV1().Pods(data.testNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "app=sftp"})
+	require.NoError(t, err)
+	for _, pod := range pods.Items {
+		for _, vm := range vmList {
+			extractPath := path.Join(uploadPath, vm.nodeName)
+			_, stderr, err := data.RunCommandFromPod(data.testNamespace, pod.Name, "", []string{"sh", "-c", fmt.Sprintf("mkdir %s", extractPath)})
+			if stderr != "" {
+				t.Errorf("error when creating folder %s on pod %s/%s: %s", extractPath, data.testNamespace, pod.Name, stderr)
+			}
+			require.NoError(t, err)
+			_, stderr, err = data.RunCommandFromPod(data.testNamespace, pod.Name, "", []string{"sh", "-c", fmt.Sprintf("tar xvf %s/%s_%s.tar.gz -C %s", uploadPath, vm.nodeName, bundleName, extractPath)})
```

Comment:
Not suggested to ignore stdout, it's helpful sometimes for troubleshooting.
Better to add it in following logs.

Reply:
updated